### PR TITLE
Adds chain link fences

### PR DESCRIPTION
**Pull request by unid15**
Like transparent walls but more fun. Can't be created - for mapping only

FEATURES:
- can be cut with wirecutters (up to 3 times, 20 seconds per cut)
- cutting once lets you pass items through the fence
- cutting twice lets you climb through the fence unless you're fat (takes 20 seconds)
- cutting thrice lets you walk through the fence
- for air and projectiles, same rules as for grilles (air can pass through, projectiles only sometimes)
- can be shocked by putting a live wire underneath them - like grilles
- clicking on them with harm intent causes you to slam the fence
- fence doors (click to open or close, not really secure)
- secure fence doors (can only be opened and closed from one side)

Featuring awful sprites

f

### CODE REVIEW

**Comment by FudgePucker** [comment]
I can't wait to make interment camps for vox.

**Comment by ComicIronic** [comment]
I see a lot of problems with how these look next to walls and windows.
Perspective resprite when

**Comment by unid15** [comment]
WYSI

**Comment by Exxion** [comment]
I'm going to have a fun time getting things with perspective like that to layer properly with pixel movement

**Comment by unid15** [comment]
it doesn't look so bad next to walls
also updated the PR with more features - see original post

File: code/game/objects/structures/fence.dm [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+				switch(++hole_size)
+					if(SMALL_HOLE)
+						visible_message("<span class='notice'>\The [user] creates a small opening in \the [src] with \the [W].</span>")
+						to_chat(user, "<span class='info'>This hole seems to be [user.is_fat() ? "way " : ""]too small to climb though, but you probably could throw something through it.</span>")
+					if(MEDIUM_HOLE)
+						visible_message("<span class='notice'>\The [user] cuts into \the [src] some more.</span>")
+						if(user.is_fat())
+							to_chat(user, "<span class='info'>While a thinner person could climb through this hole, it's still too small for you.</span>")
+						else
+							to_chat(user, "<span class='info'>You could probably fit yourself through that hole now. Although it would be better if you made it slightly bigger.</span>")
```
Comment (ghost) [inline code/game/objects/structures/fence.dm]: Why would it be better?

Comment (unid15) [inline code/game/objects/structures/fence.dm]: Takes 20 seconds to climb through a medium hole, while climbing through a large hole is instant

**Comment by jakmak** [comment]
Very nice and can be used for aesthetic vaults/gateways

File: code/game/objects/structures/fence.dm (hunk 1 above)
Comment (ghost) [inline code/game/objects/structures/fence.dm]: But it takes 20 seconds to cut a medium hole into a large hole.

**Comment by Mack3r3l** [comment]
Can these be used to slow down Kudzu vines somehow?

**Comment by AlphaBuffalo** [comment]
If you have a north south fence how will you be able to tell of there is a hole or a gate if it just looks like a line?

File: code/game/objects/structures/fence.dm (hunk 1 above)
Comment (Exxion) [inline code/game/objects/structures/fence.dm]: 20 seconds seems a bit long for climbing through
And Celt, it takes 20 seconds to climb through _each time_, whereas cutting it again is 20 seconds once

**Comment by unid15** [comment]
Here's how a gate looks



Here are the holes

File: code/game/objects/structures/fence.dm (hunk 1 above)
Comment (ghost) [inline code/game/objects/structures/fence.dm]: I just don't like being lectured about what would be "better" by the game.

Comment (ghost) [inline code/game/objects/structures/fence.dm]: ESPECIALLY when it being better is subjective - what if you're being chased?

Comment (Exxion) [inline code/game/objects/structures/fence.dm]: In that case you're fucked either way
But there's essentially no difference in time

Also shut up this is fine and you know it

Comment (ghost) [inline code/game/objects/structures/fence.dm]: no it isn't fine the game should never offer opinions on what you are doing, just give feedback.

**Comment by Jalleo** [comment]
I think these are the same sprites fallout13 uses.

**Comment by clusterfack** [comment]
I was pretty sure they had come from there too, not that I mind at all

**Comment by unid15** [comment]
I sprited them myself actually

**Comment by clusterfack** [comment]
:astonished:

**Comment by clusterfack** [comment]
Hahaha our perspective is so fucked :joy:

**Pull request merged**